File: app/src/main/java/app/we/go/emojidraw/arch/di/features/practice/PracticeConfigurationModule.java
package app.we.go.emojidraw.arch.di.features.practice;

import app.we.go.emojidraw.arch.di.qualifier.PracticeDuration;
import app.we.go.emojidraw.arch.di.qualifier.PracticeSize;
import app.we.go.emojidraw.arch.di.scope.ActivityScope;
import dagger.Module;
import dagger.Provides;

@ActivityScope
@Module
public class PracticeConfigurationModule {

    @Provides
    @ActivityScope
    @PracticeDuration
    static Integer providePracticeDuration() {
        return 60;
    }

    @Provides
    @PracticeSize
    @ActivityScope
    static Integer providePracticeSize() {
        return 10;
    }
}

File: app/src/main/java/app/we/go/emojidraw/arch/di/features/practice/PracticeConfigurationModule.kt
package app.we.go.emojidraw.arch.di.features.practice

import app.we.go.emojidraw.arch.di.PracticeDuration
import app.we.go.emojidraw.arch.di.ActivityScope
import dagger.Module
import dagger.Provides

@Module
object PracticeConfigurationModule {

    @Provides
    @JvmStatic
    @ActivityScope
    @PracticeDuration
    fun providePracticeDuration(): Int {
        return 60
    }

    @Provides
    @JvmStatic
    @ActivityScope
    fun providePracticeSize(): Int {
        return 10
    }
}
